crates/typst/src/model/reference.rs

```diff
@@ -95,22 +110,32 @@ use crate::text::TextElem;
 pub struct RefElem {
     /// The target label that should be referenced.
     ///
-    /// Can be a label that is defined in the document or an entry from the
+    /// Can be a label that is defined in the document or, if the
+    /// [`form`]($ref.form) is set to `["normal"]`, an entry from the
     /// [`bibliography`].
     #[required]
     pub target: Label,
 
     /// A supplement for the reference.
     ///
-    /// For references to headings or figures, this is added before the
-    /// referenced number. For citations, this can be used to add a page number.
+    /// If the [`form`]($ref.form) is set to `{"normal"}`:
+    /// - For references to headings or figures, this is added before the
+    ///   referenced number.
+    /// - For citations, this can be used to add a page number.
+    ///
+    /// If the [`form`]($ref.form) is set to `{"page"}`, then this is added
+    /// before the page number of the label referenced.
     ///
-    /// If a function is specified, it is passed the referenced element and
-    /// should return content.
+    /// If a function is specified, it is passed both the referenced element and
+    /// the reference [`form`]($ref.form), and should return content.
     ///
     /// ```example
+    /// #set page(numbering: "1")
     /// #set heading(numbering: "1.")
-    /// #set ref(supplement: it => {
+    /// #set ref(supplement: (it, form) => {
+    ///   if form == "page" {
+    ///     return "p."
+    ///   }
```

Comment:
I think there's no need to pass `form` here as you can just write `show ref.where(form: "page"): set ref(supplement: ..)`. That way, it's no breaking change.

In fact, I'd like to get rid of allowing a function in general in the future, but that's left for later.

Reply:
Good point, I'll update it.

I'd have thought functions might be a good way to go about solving https://github.com/typst/typst/issues/2485, though that then wouldn't really fit the `supplement` parameter I guess.

Reply:
Is there a nice where to break the line `show ref.where(form: "page"): \ set ref(supplement: ..)`? The compiler complains about expected expression, and nesting another function isn't great either.

Reply:
There is not.

Reply:
```
show ref.where(
  form: "page"
): set ref(supplement: ..)
```
is the best you can do

Reply:
it is sometimes used in the docs I think

Reply:
Okay, I'll do that then.